tools/codegen/gen_backend_stubs.py

```diff
@@ -78,7 +83,7 @@ def create_backend_index(backend_ops: List[str], dispatch_key: DispatchKey) -> B
         with context(lambda: f'The provided value for "backend" must be a valid DispatchKey, but got {backend}.'):
             backend_key = DispatchKey.parse(backend)
 
-        backend_idx = create_backend_index(supported, backend_key)
+        backend_idx = create_backend_index(supported, backend_key, use_out_as_primary=use_out_as_primary)
         assert backend_key not in backend_indices
```

Comment:
one open:
i use pudb to debug python program, i usually restart the program within pudb, but it hits the assert here in the second run. The reason is that _GLOBAL_PARSE_NATIVE_YAML_CACHE is not clear in the beginning of the second run, and so the backend_key is already in backend_indices. Any comment? thanks.

Reply:
cc @albanD. We added the cache because it gives a pretty large speed-up to the build, but this is probably a downside. (Although hopefully running the backend codegen multiple times in the python runtime isn't something we have to do a lot). You can probably either explicitly clear the cache in the pudb session (?), or just restart pudb.

Reply:
We do have the assumption that the codegen is a fresh python run everytime yes. 
You can get away with not restarting by clearing the cache etc (as we do in some testing) but that is brittle at best.

Reply:
thanks both.